src/omnicore/omnicore.cpp

```diff
@@ -784,6 +784,7 @@ static int parseTransaction(bool bRPConly, const CTransaction& wtx, int nBlock,
     std::vector<std::string> script_data;
     std::vector<std::string> address_data;
     std::vector<int64_t> value_data;
+    uint16_t nulldata_vout = 0;
```

Comment:
The protocol supports multiple OP_RETURN outputs sure, but it's not standard and it's not realistic to expect to get one into a block (unless you're a miner).  Reality on the ground is most miners use the default policies, so I kind of see supporting multiple OP_RETURN outputs when they can't be used on the underlying transport network as a little pointless.

That's just an explanation of my reasoning for putting it this change in though.  I'm happy to remove if you'd prefer to keep the multiple nulldata support in place, this change doesn't offer as much improvement as some of the others.   However just out of interest, have you ever actually tested the rest of the program with multiple OP_RETURN outputs eg on regtest?  I suspect that this isn't the only place an assumption that there is only one nulldata output exists has been made (at least by me hehe).